lib/ood_core/job/task.rb

```diff
@@ -0,0 +1,24 @@
+module OodCore
+  module Job
+    class Task
+      attr_reader :id
+      attr_reader :status
+
+      def initialize(id:, status:, **_)
+        @task_id = id
+        @status = OodCore::Job::Status.new(state: status)
+      end
+
+      def to_h
+        {
+          :id => @id,
+          :status => @status
```

Comment:
best practice to use the getters here instead of the instance vars directly:

```ruby
def to_h
  {
    :id => id,
    :status => status
  }
end
```